tooling/azure-automation/resources-cleanup/src/test_resources_cleanup.py

```diff
@@ -125,4 +129,195 @@ def test_get_dry_run_default(monkeypatch, capsys):
     monkeypatch.delenv("DRY_RUN", raising=False)
     assert get_dry_run() is False
     captured = capsys.readouterr()
-    assert "Info: DRY_RUN not set in automation variables or environment" in captured.out
\ No newline at end of file
+    assert "Info: DRY_RUN not set in automation variables or environment" in captured.out
+
+
+# Tests for get_boolean_from_string function
```

Comment:
Is it really this important to allow users to pass any of these? Why? Support `'true'` and `'false'` and call it a day.